src/components/project/project.service.ts

```diff
@@ -582,6 +584,7 @@ export class ProjectService {
         value: result.owningOrganizationId,
       },
       canDelete: await this.db.checkDeletePermission(id, sessionOrUserId),
+      pinned: await this.pinSerivce.isPinned(id, sessionOrUserId as Session), // TODO: More safe way to pass session
```

Comment:
@CarsonF What is the best way to pass the session here?

Reply:
Can we match the relation in the same query instead of doing another lookup?